---
int64 numBricks
BlockParams[] poses